test-programs/rnd-inf-test/tests/test.rs

```diff
@@ -0,0 +1,932 @@
+        let nullifier_queue_keypair = Keypair::new(); //from_seed(&[self.rng.gen_range(0..255); 32]).unwrap();
+        create_address_merkle_tree_and_queue_account(
+            &self.payer,
+            &mut self.context,
+            &merkle_tree_keypair,
+            &nullifier_queue_keypair,
+            None,
+            self.indexer.address_merkle_trees.len() as u64,
+        )
+        .await;
+        let init_value = BigUint::from_str_radix(
+            &"21888242871839275222246405745257275088548364400416034343698204186575808495616",
+            10,
+        )
+        .unwrap();
+        let mut merkle_tree = Box::new(
+            IndexedMerkleTree::<Poseidon, usize>::new(
+                STATE_MERKLE_TREE_HEIGHT as usize,
+                STATE_MERKLE_TREE_CANOPY_DEPTH as usize,
+            )
+            .unwrap(),
+        );
+        let mut indexed_array =
+            Box::new(IndexedArray::<Poseidon, usize, INDEXED_ARRAY_SIZE>::default());
+
+        merkle_tree.append(&init_value, &mut indexed_array).unwrap();
+        self.indexer
+            .address_merkle_trees
+            .push(AddressMerkleTreeBundle {
+                accounts: AddressMerkleTreeAccounts {
+                    merkle_tree: merkle_tree_keypair.pubkey(),
+                    queue: nullifier_queue_keypair.pubkey(),
+                },
+                merkle_tree,
+                indexed_array,
+            });
+        // TODO: Add assert
+    }
+
+    pub fn safe_gen_range<T, R>(rng: &mut StdRng, range: R, empty_fallback: T) -> T
+    where
+        T: SampleUniform + Copy,
+        R: SampleRange<T> + Sized,
+    {
+        if range.is_empty() {
+            return empty_fallback;
+        }
+        rng.gen_range(range)
+    }
+
+    /// 1. Transfer spl tokens between random users
+    pub async fn activate_keypair_actions(&mut self, user: &Pubkey) {
+        let user_index = self
+            .users
+            .iter()
+            .position(|u| &u.keypair.pubkey() == user)
+            .unwrap();
+        // compress spl
+        // check sufficient spl balance
+        if self
+            .rng
+            .gen_bool(self.keypair_action_config.decompress_spl.unwrap_or(0.0))
+            && self.users[user_index].token_accounts.len() != 0
+        {
+            println!("\n --------------------------------------------------\n\t\t Compress Spl\n --------------------------------------------------");
+            let mut balance = 0;
+            let mut mint = Pubkey::default();
+            let mut token_account = Pubkey::default();
+            for _ in 0..self.users[user_index].token_accounts.len() {
+                let (_mint, _token_account) = self.users[user_index].token_accounts[self
+                    .rng
+                    .gen_range(0..self.users[user_index].token_accounts.len())];
+                token_account = _token_account;
```

Comment:
Just wondering - do you expect to have the token accounts also somehow randomized?

So far, from my understanding, this loop will always take the first token account with positive balance.

I see that later, we are using this utility to pick a randomized SPL token account

```rust
let (mint, token_accounts) = self.select_random_spl_token_accounts(user).await;
```

Would that be applicable here, and in other places using the 1st non-empty token account, as well? I see that `self.select_random_spl_token_accounts` already filters for non-empty.

Reply:
> do you expect to have the token accounts also somehow randomized?
every iteration selects a token account randomly by generating a random index with:
`` self.rng.gen_range(0..self.users[user_index].token_accounts.len())``
is this what you mean?

yeah it could be a method

Reply:
wrt ``select_random_spl_token_accounts``
naming is off it should be named ``select_random_compressed_token_accounts``